Eval for Ukrainian electronic petitions

ziomio opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, pelase note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑
### Eval name
Ukrainian electronic petitions

### Eval description
This eval provides electronic petitions summaries of ukrainian citizens to the government submitted through https://petition.president.gov.ua/?status=processed website and that have government reply in Accepted/Declined form.


### What makes this a useful eval?
To correctly answer "Accepted" or "Declined" GPT-4 or future LLMs need to know Ukrainian laws (civilian, criminal, administrative and many others). This eval is useful because it could help to predict government reply right now, without waiting for months until petition gains minimally required amount of votes (usually it's thousands of votes) to be marked as "Will be reviewed by government".

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

It is a high quality eval because I manually analyzed and summarized texts from all examples and then read government's very long reply texts to understand if they Accepted or Declined those petitions.

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Повернути пільги на оплату електроенергії для населення, яке проживає у 30-кілометровій зоні атомних електростанцій.  Відновити інші пільги населенню на оплату спожитої електричної енергії. Дата складання петиції - 2021 рік."}], "ideal": "Уряд не підтримав цю петицію, оскільки вважає, що питання надання пільг та субсидій на оплату комунальних послуг, в тому числі електроенергії, є повноваженням Кабінету Міністрів України а не уряду. Крім того, уряд вважає, що в державному бюджеті України на 2021 рік вже передбачено достатній обсяг коштів для компенсації витрат вразливим верствам населення, у тому числі для населення, що проживає в 30-кілометровій зоні атомних електростанцій."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Воїн-герой, який загинув у бою під час захисту України, заслуговує на звання Героя України."}], "ideal": "Уряд не підтримав цю петицію, бо відповідно до законодавства лише Президент України має повноваження присвоювати звання Героя України. Уряд не має таких повноважень і не може безпосередньо задовольнити цю петицію."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Встановити системи відеоспостереження в дошкільних закладах освіти для запобігання жорстокому поводженню з дітьми. Пропонують розробити законопроект, який зобов\\'язує дошкільні заклади освіти встановлювати системи відеоспостереження."}], "ideal": "Уряд не підтримав цю петицію, оскільки вважає, що питання освіти і забезпечення безпеки дітей у дошкільних закладах належить до компетенції Кабінету Міністрів України. Таким чином, уряд передав це питання на розгляд прем'єр-міністру для подальшого вивчення і розробки заходів з урахуванням конституційних прав громадян."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Дозволити виїзд студентам-чоловікам, які навчаються за кордоном. Загальна мобілізація обмежує право на виїзд з країни. Звернення було написано в січні 2023 року."}], "ideal": "Уряд не підтримав петицію, бо в Україні діяв воєнний стан і загальна мобілізація. Згідно із законодавством, під час цих надзвичайних ситуацій дозволяється обмежувати деякі конституційні права громадян, зокрема свободу пересування."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Відмінити примусову вакцинацію населення. Людей не можна примушувати вакцинуватися простроченими вакцинами під страхом покарання"}], "ideal": "Уряд не підтримав цю петицію, оскільки закони України передбачають обов'язковість деяких щеплень для запобігання поширенню небезпечних інфекційних захворювань. Уряд зазначив, що відповідно до Конституції України, законів «Про захист населення від інфекційних захворювань» та «Про забезпечення санітарного та епідемічного благополуччя населення» передбачено обов'язковість деяких щеплень."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Підвищити заробітну плату пожежним, які, рятуючи наше життя, жертвують власним. Пожежні заслуговують на гідну заробітну плату для гідного життя."}], "ideal": "Уряд підтримав цю петицію, оскільки він погоджується з тим, що заробітна плата пожежних потребує перегляду, щоб забезпечити їм гідне життя, враховуючи небезпечні умови їх роботи. При цьому уряд зазначає, що вже зроблено певні кроки для поліпшення соціального захисту пожежних, але комплексне вирішення порушених у петиції питань належить до повноважень Кабінету Міністрів України."}
{"input": [{"role": "system", "content": "Ви отримаєте текст електронної петиції. Вам потрібно проаналізувати суть звернення та опираючись на законодавчу базу України та інші фактори відповісти чи підтримали би уряд цю петицію. Поясніть свій хід думок та висновок з позиції законодавства України."}, {"role": "user", "content": "Суть звернення: Впровадити обов\\'язковий курс статевого виховання у школах та вишах. Статеве виховання дітей і молоді значно знижує ризики насильства і підвищує відповідальність."}], "ideal": "Уряд не підтримав цю петицію, оскільки вважає, що питання впровадження обов'язкового статевого виховання у школах є прерогативою Верховної Ради України та Кабінету Міністрів України. Відповідно до Конституції України Президент не має повноважень приймати такі рішення, а лише може звертатися з проханням до КМУ розглянути це питання."}

  ```
</details>

luqman-openai reviewed (changes requested):
Thanks for the contribution. I would like to request some changes.

I would recommend using the `ModelBasedClassify` with `eval_type: cot_classify`  and with the `fact` spec, instead of `Match` method here. Asking the model to provide reasoning will give the model a fair chance to solve the question, and a proper evaluation method can be used to evaluate the provided answer. Switching to model graded eval will require a rewrite of ideal responses with a reasonable details because the model graded classifier compares the ideal response with the model generated response.

We would love to review the PR again after the suggested changes.

ziomio commented:
@luqman-openai, thank you for suggestions. The requested changes have been made:
*  `Match` method was changed to `ModelBasedClassify` with `eval_type: cot_classify` with the `fact` spec 
* I have analyzed the reasoning behind each government's decision and updated `ideal responses` with expanded answers.
* I have updated eval samples in "Eval JSON data" section from PR template above.

Please let me know if additional changes are needed.

luqman-openai reviewed (approved):
Thanks for submitting this eval! I am approving this PR.

ziomio commented:
Dear @luqman-openai may I know when GPT-4 API access will be enabled? Five days has passed and no access yet.
